pkg/scene/widget_align_system.go

```diff
@@ -0,0 +1,250 @@
+}
+
+// Area will return a specific area in the align system.
+func (was *WidgetAlignSystem) Area(zone, section, area string) *WidgetArea {
+	if was == nil {
+		return nil
+	}
+
+	s := was.Section(zone, section)
+
+	switch area {
+	case "top":
+		return &s.top
+	case "middle":
+		return &s.middle
+	case "bottom":
+		return &s.bottom
+	}
+	return nil
+}
+
+// WidgetIds will return a slice of widget ids from a specific area.
+func (was *WidgetAlignSystem) WidgetIDs(z, s, a string) []*id.WidgetID {
+	area := was.Area(z, s, a)
+	return area.widgetIds
+}
+
+// Alignment will return the alignment of a specific area.
+func (was *WidgetAlignSystem) Alignment(z, s, a string) *string {
+	area := was.Area(z, s, a)
+	return &area.align
+}
+
+// Add a widget to the align system.
+func (was *WidgetAlignSystem) Add(wid id.WidgetID, l *WidgetLocation) {
+	if was == nil {
+		return
+	}
+
+	a := was.Area(l.Zone, l.Section, l.Area)
+	nIds := a.widgetIds
+
+	if _, b := a.Has(wid); !b {
+		nIds = append(a.widgetIds, &wid)
+	}
+	a.widgetIds = nIds
+
+	if a.align == "" {
+		a.align = "start"
+	}
+}
+
+// Remove a widget from the align system.
+func (was *WidgetAlignSystem) Remove(wid id.WidgetID) {
+	if was == nil {
+		return
+	}
+	var nwids []*id.WidgetID
+	i, loc := was.FindWidgetIDLocation(wid)
+	if loc != nil {
+		a := was.Area(loc.Zone, loc.Section, loc.Area)
+		if len(a.widgetIds) > 0 {
+			nwids = append(a.widgetIds[:*i], a.widgetIds[*i+1:]...)
+		}
+		a.widgetIds = nwids
+	}
+}
+
+// Update a widget in the align system.
+func (was *WidgetAlignSystem) Update(wid id.WidgetID, l *WidgetLocation, index *int, align *string) {
+	if was == nil {
+		return
+	}
+
+	i, oldL := was.FindWidgetIDLocation(wid)
+	a := was.Area(oldL.Zone, oldL.Section, oldL.Area)
+
+	if align != nil {
+		switch *align {
+		case AlignStart:
+			a.align = AlignStart
+		case AlignCenter:
+			a.align = AlignCenter
+		case AlignEnd:
+			a.align = AlignEnd
+		default:
+			a.align = AlignStart
+		}
+	} else {
+		a.align = AlignStart
+	}
+
+	if index != nil {
+		moveInt(a.widgetIds, *i, *index)
+	}
+	if l != nil {
+		was.Remove(wid)
+		was.Add(wid, l)
+	}
+}
+
+// moveInt moves a widget's index.
+func moveInt(array []*id.WidgetID, srcIndex int, dstIndex int) []*id.WidgetID {
+	value := array[srcIndex]
+	return insertInt(removeInt(array, srcIndex), value, dstIndex)
+}
+
+// insertInt is used in moveInt to add the widgetID to a new position(index).
+func insertInt(array []*id.WidgetID, value *id.WidgetID, index int) []*id.WidgetID {
+	return append(array[:index], append([]*id.WidgetID{value}, array[index:]...)...)
+}
+
+// removeInt is used in moveInt to remove the widgetID from original position(index).
+func removeInt(array []*id.WidgetID, index int) []*id.WidgetID {
+	return append(array[:index], array[index+1:]...)
+}
+
+// WidgetAreaFrom will add a slice of widget ids to a specific area of an align system.
+func (was *WidgetAlignSystem) AddAll(wids []*id.WidgetID, align, z, s, a string) {
```

Comment:
```suggestion
func (was *WidgetAlignSystem) AddAll(wids []*id.WidgetID, align string, loc *WidgetLocation) {
```

This is exactly where WidgetLocation should be used